beater/request/context.go

```diff
@@ -55,11 +57,11 @@ type Context struct {
 func (c *Context) Reset(w http.ResponseWriter, r *http.Request) {
 	c.Request = r
 	c.Logger = nil
-	c.TokenSet = false
-	c.Authorized = false
-	c.IsRum = false
 	c.RateLimiter = nil
+	c.Authorization = &authorization.AllowAuth{}
```

Comment:
trying to connect the dots: does https://github.com/elastic/apm-server/pull/3004/files#diff-3f6ee316627b52f99975b03690d74fb1R49 need to be run after this for every request? is that happening?

Reply:
No, [`handler.AuthorizedFor`](https://github.com/elastic/apm-server/pull/3004/files#diff-3f6ee316627b52f99975b03690d74fb1R89) is called for every request, as the actual auth method and the api_key can only be determined on a per request base.

You see the call in the [middleware.authorization](https://github.com/elastic/apm-server/pull/3004/files#diff-854a750f79bb08e98ddc18772ff72be1R34)